### PR TITLE
ft:seller delete specific Item in collection

### PR DESCRIPTION
**Pull request by emilenfc**
# Description

In this feature, a seller sould delete a specific Item/product from collection

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have created a loom video for the new feature or enhancement 
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shyakadavis — DISMISSED** [review]
LGTM

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Review by freddykaberuka — DISMISSED** [review]

**Review by musllim — DISMISSED** [review]
this looks good

File: src/controllers/Product.controller.ts [hunk 1]
```diff
@@ -81,3 +85,44 @@ export const getAvailableProducts = async (req: Request, res: Response) => {
...
+    if (error instanceof Error) {
+      console.log(error.message);
+      res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error while clearing product",
+        error: error.message,
+      });
+    } else {
+      console.log(`Unexpected error in product deleting: `, error);
```
Comment (mucyoalexaxel) [inline src/controllers/Product.controller.ts:125]: use appropiate response messages

File: src/routes/product.routes.ts [hunk 2]
```diff
@@ -19,4 +23,10 @@ productRouter.get("/available", getAvailableProducts);
 // search for products
 productRouter.get("/search", searchProducts); // search for products
 
+//delete one product
```
Comment (mucyoalexaxel) [inline src/routes/product.routes.ts:26]:
```suggestion
```

**Review by mucyoalexaxel — CHANGES_REQUESTED** [review]
Check the requested changes as well.

**Review by musllim — APPROVED** [review]
this works well

**Review by mucyoalexaxel — APPROVED** [review]
LGTM

**Review by mucyoalexaxel — APPROVED** [review]
LGTM

**Review by ISHIMWE-Aime — APPROVED** [review]
LGTM

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Review by musllim — APPROVED** [review]
looks good

**Comment by musllim** [comment]
I think you have to first fix those hound errors

**Review by shyakadavis — COMMENTED** [review]
I think what you can do is rebase from main, `again`, and see if Hound CI because aware of the version of the Ts linter being used.

File: .hound.yml [hunk 3]
```diff
@@ -2,4 +2,4 @@ eslint:
   enabled: true
   config_file: .eslintrc.json
   extensions:
-    - .ts
+    - .tsgit
```
Comment (mucyoalexaxel) [inline .hound.yml:5]:
```suggestion
    - .ts
```

**Review by mucyoalexaxel — CHANGES_REQUESTED** [review]

File: .hound.yml (hunk 3 above)
Comment (emilenfc) [reply .hound.yml]: ooo sorry,  and thank

**Review by shyakadavis — APPROVED** [review]
LGTM 🎉

**Pull request merged**